subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -260,71 +260,70 @@ static void avdtp_set_status(struct bt_avdtp_req *req, struct net_buf *buf, uint
-			err = session->ops->discovery_ind(session, &error_code);
-		}
+	int err;
+	struct bt_avdtp_sep *sep;
+	struct net_buf *rsp_buf;
+	uint8_t error_code = 0;
 
-		rsp_buf =
-			avdtp_create_reply_pdu(err ? BT_AVDTP_REJECT : BT_AVDTP_ACCEPT,
-					       BT_AVDTP_PACKET_TYPE_SINGLE, BT_AVDTP_DISCOVER, tid);
-		if (!rsp_buf) {
-			return;
-		}
+	if (session->ops->discovery_ind == NULL) {
+		err = -ENOTSUP;
+	} else {
+		err = session->ops->discovery_ind(session, &error_code);
+	}
 
-		if (err) {
-			if (error_code == 0) {
-				error_code = BT_AVDTP_BAD_STATE;
-			}
-			LOG_DBG("discover err code:%d", error_code);
-			net_buf_add_u8(rsp_buf, error_code);
-		} else {
-			struct bt_avdtp_sep_data sep_data;
-
-			SYS_SLIST_FOR_EACH_CONTAINER(&seps, sep, _node) {
-				memset(&sep_data, 0, sizeof(sep_data));
-				sep_data.inuse = sep->sep_info.inuse;
-				sep_data.id = sep->sep_info.id;
-				sep_data.tsep = sep->sep_info.tsep;
-				sep_data.media_type = sep->sep_info.media_type;
-				net_buf_add_mem(rsp_buf, &sep_data, sizeof(sep_data));
-			}
-		}
+	rsp_buf = avdtp_create_reply_pdu(err ? BT_AVDTP_REJECT : BT_AVDTP_ACCEPT,
+					 BT_AVDTP_PACKET_TYPE_SINGLE, BT_AVDTP_DISCOVER, tid);
+	if (!rsp_buf) {
+		return;
+	}
 
-		err = bt_l2cap_chan_send(&session->br_chan.chan, rsp_buf);
-		if (err < 0) {
-			net_buf_unref(rsp_buf);
-			LOG_ERR("Error:L2CAP send fail - result = %d", err);
-			return;
+	if (err) {
+		if (error_code == 0) {
+			error_code = BT_AVDTP_BAD_STATE;
 		}
+		LOG_DBG("discover err code:%d", error_code);
+		net_buf_add_u8(rsp_buf, error_code);
 	} else {
-		struct bt_avdtp_req *req = session->req;
+		struct bt_avdtp_sep_data sep_data;
 
-		if (req == NULL) {
-			return;
-		}
-		k_work_cancel_delayable(&session->timeout_work);
-		if (msg_type == BT_AVDTP_ACCEPT) {
-			DISCOVER_REQ(req)->buf = buf;
-		} else {
-			DISCOVER_REQ(req)->buf = NULL;
-		}
-		avdtp_set_status(req, buf, msg_type);
-		bt_avdtp_clear_req(session);
-		if (req->func != NULL) {
-			req->func(req);
+		SYS_SLIST_FOR_EACH_CONTAINER(&seps, sep, _node) {
+			memset(&sep_data, 0, sizeof(sep_data));
+			sep_data.inuse = sep->sep_info.inuse;
+			sep_data.id = sep->sep_info.id;
+			sep_data.tsep = sep->sep_info.tsep;
+			sep_data.media_type = sep->sep_info.media_type;
+			net_buf_add_mem(rsp_buf, &sep_data, sizeof(sep_data));
 		}
 	}
+
+	err = bt_l2cap_chan_send(&session->br_chan.chan, rsp_buf);
+	if (err < 0) {
+		net_buf_unref(rsp_buf);
+		LOG_ERR("Error:L2CAP send fail - result = %d", err);
+		return;
+	}
+}
+
+static void avdtp_discover_rsp(struct bt_avdtp *session, struct net_buf *buf, uint8_t msg_type)
+{
+	struct bt_avdtp_req *req = session->req;
+
+	if (req == NULL) {
+		return;
+	}
+	k_work_cancel_delayable(&session->timeout_work);
+	if (msg_type == BT_AVDTP_ACCEPT) {
+		DISCOVER_REQ(req)->buf = buf;
+	} else {
+		DISCOVER_REQ(req)->buf = NULL;
+	}
+	avdtp_set_status(req, buf, msg_type);
+	bt_avdtp_clear_req(session);
+	if (req->func != NULL) {
+		req->func(req);
+	}
```

Comment:
This could use a bit more vertical whitespace to improve readability. E.g. add empty lines after `}` and before `if (...`